ZFSin/zfs/module/zfs/zfs_vnops_windows.c

```diff
@@ -4700,6 +4706,7 @@ fsDispatcher(
 		break;
 	}
 
+	FsRtlExitFileSystem();
 
 	/* Re-check (since MJ_CREATE/vnop_lookup might have set it) vp here, to see if
```

Comment:
you should move the FsRtlExitFileSystem() call after the re-check since the re-chack is also touching filesystem structures.